### PR TITLE
ENH: Update compiler versions for C++14

### PR DESCRIPTION
**Pull request by N-Dekker**
Following pull request https://github.com/InsightSoftwareConsortium/ITK/pull/2563
"ENH: Upgrade ITK from C++11 to C++14"

Suggested by Jon Haitz Legarreta Gorroño (@jhlegarreta) at https://github.com/InsightSoftwareConsortium/ITK/pull/2563/#issuecomment-852134729

### CODE REVIEW

File: SoftwareGuide/Latex/Introduction/Installation.tex
Comment (jhlegarreta) [inline SoftwareGuide/Latex/Introduction/Installation.tex:37]: VS 2015 might need to be revised. Don't know about the toolsets.

**Review by jhlegarreta — COMMENTED** [review]
Looks good to me. I'd say that wisest would be to wait until the changes are agreed upon and merged in ITK before merging.

**Comment by N-Dekker** [comment]
Note to maintainers: this pull request should _only_ be merged once ITK has indeed moved from C++11 to C++14 (possibly by pull request InsightSoftwareConsortium/ITK#2563 ), as commented by @jhlegarreta

File: SoftwareGuide/Latex/Introduction/Installation.tex
Comment (N-Dekker) [reply SoftwareGuide/Latex/Introduction/Installation.tex:37]: > VS 2015 might need to be revised.

Thanks @jhlegarreta It's not strictly necessary to drop VS 2015, as it already supports most of C++14. But then again, it does still miss some C++14 features (specifically [Extended constexpr](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3652.html)).  So it's still to be discussed: https://discourse.itk.org/t/may-we-start-using-c-14-at-the-master-branch/4172/2

**Comment by thewtex** [comment]
Thanks for adding this @N-Dekker !

**Comment by N-Dekker** [comment]
@thewtex You're welcome, but I'm not entirely sure if my pull request is OK. Looking at: 
https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/blob/ae7db1ae64586ec6183f4d642787ba4b7d1c1330/SoftwareGuide/Latex/Introduction/Installation.tex#L32-L40

Versus:

https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/SupportedCompilers.md 

Should they be the same? Because now they are different! Even after my pull request 🤔 

Apparently ITK already required AppleClang 7.0.2 and later (from Xcode 7.2.1) before the upgrade to C++14! Should that requirement still be adjusted for C++14?

File: SoftwareGuide/Latex/Introduction/Installation.tex
Comment (N-Dekker) [reply SoftwareGuide/Latex/Introduction/Installation.tex]: VS 2015 dropped with my latest amend + [force-push](https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/compare/fc64bda5a4a987a736a710f611e7814ed752b20c..80b8f8913eabd3c4d46a64389e0e1bd442e009da).

**Comment by thewtex** [comment]
> Should they be the same? Because now they are different! 

Yes, could they please be synchronized?

**Comment by N-Dekker** [comment]
@thewtex 

> Yes, could they please be synchronized?

They may not mean exactly the same thing: The compilers that "should have nightly dashboard" and the "supported" compilers. Or should they be exactly the same compilers?

Anyway, this pull request does already synchronize MSVC and GCC. For (Apple) Clang I just do not know the preferred minimum. We  AppleClang 12.0.0.12000032 Xcode 12.4 (on GitHub Actions) and Apple clang version 12.0.5 (clang-1205.0.22.9) (as @ViktorvdValk told me). Sean (@seanm) is ok with Xcode 7.2.1 being the minimum.

**Comment by seanm** [comment]
You may as well put Xcode 7.2.1 as the minimum.

It's the newest possible Xcode on macOS 10.10.x, which is the oldest version of macOS that I have a bot for.  I don't think anyone else runs any Mac bot older than that.

**Comment by N-Dekker** [comment]
> You may as well put Xcode 7.2.1 as the minimum.

Thanks @seanm Please check https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/pull/156/commits/f65f786d883a4de033da118908cf06aacb9e7c2b !

**Comment by thewtex** [comment]
> They may not mean exactly the same thing: The compilers that "should have nightly dashboard" and the "supported" compilers. Or should they be exactly the same compilers?

They should be the same compilers. We likely could use more dashboard builds, but that is a separate issue.

**Review by seanm — APPROVED** [review]

**Review by thewtex — APPROVED** [review]

**Pull request merged**
